transport/oob/oob.go

```diff
@@ -0,0 +1,137 @@
+package oob
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"syscall"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type oobWriter struct {
+	writer      io.Writer
+	oobPosition int64
+	oobByte     byte // Byte to send as OOB
+	disOOB      bool // Flag to enable disOOB mode
+}
+
+var _ io.Writer = (*oobWriter)(nil)
+
+type oobWriterReaderFrom struct {
+	*oobWriter
+	rf io.ReaderFrom
+}
+
+var _ io.ReaderFrom = (*oobWriterReaderFrom)(nil)
+
+// NewOOBWriter creates an [io.Writer] that sends an OOB byte at the specified "oobPosition".
+// If disOOB is enabled, it will apply the --disOOB strategy.
+// "oobByte" specifies the value of the byte to send out-of-band.
+func NewOOBWriter(writer io.Writer, oobPosition int64, oobByte byte, disOOB bool) io.Writer {
```

Comment:
This can't be any writer, it must be a *TCPConn. Let's make that explicit.

Reply:
added